app/assets/scripts/components/csv-btn.js

```diff
@@ -0,0 +1,160 @@
+    }
+
+    if (this.props.categorySummary) {
+      let categorySummary = serialize.summary(this.props.categorySummary);
+      csv += '\n\nCategory Summary\n\n';
+      csv = csv.concat(serialize.serialize(categorySummary));
+    }
+
+    if (this.props.categoryFunds && this.props.categoryFunds.length) {
+      let categoryFunds = serialize.chartData(this.props.categoryFunds);
+      csv += '\n\nCategory Funding for Projects (US dollars)\n\n';
+      csv = csv.concat(serialize.serialize(categoryFunds));
+    }
+
+    if (this.props.categoryCompletion && this.props.categoryCompletion.length) {
+      let categoryCompletion = serialize.chartData(this.props.categoryCompletion);
+      csv += '\n\nCategory Percentage Complete By Project\n\n';
+      csv = csv.concat(serialize.serialize(categoryCompletion));
+    }
+
+    if (this.props.categoryProjectCountCompare && this.props.categoryProjectCountCompare.length) {
+      let categoryProjectCounts = serialize.chartData(this.props.categoryProjectCountCompare);
+      csv += '\n\nComparison of Number of Projects per Category\n\n';
+      csv = csv.concat(serialize.serialize(categoryProjectCounts));
+    }
+
+    if (this.props.categoryProjectFundsCompare && this.props.categoryProjectFundsCompare.length) {
+      let categoryProjectFunds = serialize.chartData(this.props.categoryProjectFundsCompare);
+      csv += '\n\nComparison of Project Funding per Category (US dollars)\n\n';
+      csv = csv.concat(serialize.serialize(categoryProjectFunds));
+    }
+
+    if (this.props.donorSummary) {
+      let donorSummary = serialize.summary(this.props.donorSummary);
+      csv += '\n\nDonor Summary\n\n';
+      csv = csv.concat(serialize.serialize(donorSummary));
+    }
+
+    if (this.props.donorProjectFunding) {
+      let donorProjectFunding = serialize.chartData(this.props.donorProjectFunding);
+      csv += '\n\nDonor Project Funding\n\n';
+      csv = csv.concat(serialize.serialize(donorProjectFunding));
+    }
+
+    if (this.props.ministrySummary) {
+      let ministrySummary = serialize.ministryOwnerSummary(this.props.ministrySummary);
+      csv += '\n\nMinistry Summary\n\n';
+      csv = csv.concat(serialize.serialize(ministrySummary));
+    }
+
+    if (this.props.ministryActiveProjects) {
+      let ministryActiveProjects = serialize.ministryActiveProjects(this.props.ministryActiveProjects);
+      csv += '\n\nActive Project Status\n\n';
+      csv = csv.concat(serialize.serialize(ministryActiveProjects));
+    }
+
+    if (this.props.servedByProject) {
+      let servedByProject = serialize.chartData(this.props.servedByProject);
+      csv += '\n\nNumber Served Per Project\n\n';
+      csv = csv.concat(serialize.serialize(servedByProject));
+    }
+
+    if (this.props.ownerSummary) {
+      let ownerSummary = serialize.ministryOwnerSummary(this.props.ownerSummary);
+      csv += '\n\nOwner Summary\n\n';
+      csv = csv.concat(serialize.serialize(ownerSummary));
+    }
```

Comment:
@nbumbarger the main issue with this block is that you repeat yourself many times. For example, there are 9(!) separate conditionals that just call `serialize.chartData` on the same data structure. This produces code that is hard to maintain.

One approach to making this a little tighter is to change the data structure and aggregate all the `chartData` calls into a single array.

```(javascript)
chartData = [
  {
    data: [/* data */],
    title: 'Number Served Per Project'
  },
  {
    data: [/* data */],
    title: 'Category Percentage Complete By Project'
  }
];
```

At this point all you need to do is loop over the array.

```(javascript)
this.props.chartData.forEach(d => {
  csv += '\n\n' + d.title + '\n\n';
  csv += serialize.serialize(serialize.chartData(d.data));
});
```

Anytime you repeat yourself this many times, I think it's worth stepping back and seeing whether you can approach the problem a slightly different way.